singularity-eos/eos/eos_variant.hpp

```diff
@@ -840,6 +840,22 @@ class Variant {
         eos_);
   }
 
+  // Tooling for modifiers
+  PORTABLE_FORCEINLINE_FUNCTION
+  bool IsModified() const {
+    return mpark::visit([](const auto &eos) { return eos.IsModified(); }, eos_);
+  }
+  PORTABLE_FORCEINLINE_FUNCTION
+  Variant UnmodifyOnce() {
+    return mpark::visit(
+        [](auto &eos) { return eos_variant<EOSs...>(eos.UnmodifyOnce()); }, eos_);
+  }
+  PORTABLE_FORCEINLINE_FUNCTION
+  Variant GetUnmodifiedObject() {
+    return mpark::visit(
+        [](auto &eos) { return eos_variant<EOSs...>(eos.GetUnmodifiedObject()); }, eos_);
+  }
```

Comment:
All EOS's are assumed to have reference semantics, including databoxes, so copying will not cause correctness problems, though it may cause performance bottlenecks.

I was not aware that one could do a type-matching visit. That's cool. There may be other uses for that.